src/xref2/strengthen.ml

```diff
@@ -20,21 +20,30 @@ open Delayed
 let rec signature :
     Cpath.module_ -> ?canonical:Cpath.module_ -> Signature.t -> Signature.t =
  fun prefix ?canonical sg ->
+  let sg', strengthened_modules = sig_items prefix ?canonical sg in
+  (* Format.eprintf "Invalidating modules: %a\n%!" (Format.pp_print_list Ident.fmt) strengthened_modules; *)
+  let substs =
+    List.fold_left
+      (fun s mid -> Subst.path_invalidate_module (mid :> Ident.path_module) s)
+      Subst.identity strengthened_modules
+  in
+  Subst.signature substs sg'
```

Comment:
We can apply this only if `strengthened_modules` is not empty.

Reply:
I couldn't measure a performance improvement (it is smaller than the noise).

Reply:
I'm slightly hesitant to do this because it would mean the identifiers don't get fresh names even though they're changed.

Reply:
I haven't thought the identity subst would still have an effect. It makes sense here.